src/main/mondrian/rolap/SqlContextConstraint.java

```diff
@@ -108,13 +120,29 @@ public static boolean isValidContext(
         // we can not handle all calc members in slicer. Calc measure and some
         // like aggregates are exceptions
         Member[] members = context.getMembers();
-        for (int i = 1; i < members.length; i++) {
-            if (members[i].isCalculated()
-                && !SqlConstraintUtils.isSupportedCalculatedMember(members[i]))
+        for (int i = 0; i < members.length; i++) {
+            boolean isCalculated = members[i].isCalculated();
+            if (isCalculated && !SqlConstraintUtils
+                .isSupportedCalculatedMember(members[i]))
             {
                 return false;
             }
         }
+        // see MONDRIAN-1999
+        int maxConstraints =
+            MondrianProperties.instance().MaxConstraints.get();
+        int supportedCalculatedMembersQuantity =
+            SqlConstraintUtils
+                .expandSupportedCalculatedMembers(
+                    members, context).length;
+        if (supportedCalculatedMembersQuantity > maxConstraints) {
```

Comment:
@mkambol, you are absolutely right. I reworked the fix in order to verify calculation members quantity for each aggregation and added appropriate tests.Thanks.